### DISCUSSION TITLE
Error trying to insert to MongoDb with an array of objects attribute

### DISCUSSION THREAD

**Discussion by Marcos-Iorio** [prisma/prisma#17712] [Q&A]
Hi folks!
I have a very huge problem that im struggling with it since 3 days ago.
I will post images of all that i will relate for you to understand.
So i have a Model called Orders where one attribute is OrderProducts[], OrdersProducts is a composite type with attributes of the products in the order(in the order may be there are more than one product so for that is an array). I have an interface in Typescript modeling the product attributes as IProduct[], when I try to create an order and pass the products to products: i receive an error with Type IProducts[]
![Typescript interfaces](https://user-images.githubusercontent.com/65930894/216426376-f37c8113-08b1-4397-9cff-a8943b28cdf8.png)
![Problem](https://user-images.githubusercontent.com/65930894/216426382-c1bd3f8a-a7f0-4dd8-b5e4-273c24cc22ff.png)
![Prisma-schemas](https://user-images.githubusercontent.com/65930894/216426387-e6bf6215-bc46-4b33-9087-f0c3a9509624.png)

![Order-Products Type](https://user-images.githubusercontent.com/65930894/216426893-4407035b-c276-48b0-bdc5-2c742db39dc1.png)
![full-error](https://user-images.githubusercontent.com/65930894/216427154-be46c870-7b4e-4b32-b6f3-e1874b6fe9fb.png)

If you need more details just tell me!

**Comment by nurul3101** [comment]
Hey @Marcos-Iorio 👋 

I tried to run the create query that you shared.
In the model `Orders`, there is a field of `total_price`, but in the query, you are passing a field `total`.

Considering this schema:
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["extendedWhereUnique"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Users {
  id     String   @id @default(auto()) @map("_id") @db.ObjectId
  email  String   @unique
  name   String
  Orders Orders[]
}

model Products {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  category    String
  name        String
  description String
  main_image  String
  images      String[]
  stock       Int
  price       Int
}

model Orders {
  id              String           @id @default(auto()) @map("_id") @db.ObjectId
  order_id        String           @unique
  user_name       String
  email           String
  products        OrdersProducts[]
  total_price     Int
  order_timestamp DateTime         @db.Date
  status          String
  address         Address
  user            Users            @relation(fields: [email], references: [email])
}

type Address {
  street  String
  city    String
  country String
}

type OrdersProducts {
  category    String
  description String
  id          String
  main_image  String
  name        String
  price       Float
  quantity    Int
}
``` 

This is the query that I tried:
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({});

async function main() {
  const response = await prisma.orders.create({
    data: {
      order_id: '1',
      user_name: 'John',
      email: 'john@prisma.io',
      products: [
        {
          category: 'Electronics',
          description: 'A cool new phone',
          id: '1',
          main_image: 'https://example.com/phone.jpg',
          name: 'Phone XL',
          price: 1000,
          quantity: 1,
        },
        {
          category: 'Electronics 2',
          description: 'A cool new phone 2',
          id: '2',
          main_image: 'https://example.com/phone.jpg',
          name: 'Phone XL 2',
          price: 1000,
          quantity: 1,
        },
      ],
      total_price: 2000,
      order_timestamp: new Date(),
      status: 'PENDING',
      address: {
        street: '123 Main St',
        city: 'San Francisco',
        country: 'USA',
      },
    },
  });

  console.log(response);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

And here's the working response:
<img width="650" alt="Screenshot 2023-02-03 at 6 52 56 PM" src="https://user-images.githubusercontent.com/13049676/216614002-62854fee-db56-499d-942c-feb5e2350952.png">

Can you check what values you pass in the `orderInfo.products` variable?

**Reply by Marcos-Iorio** [reply]
Hi Nurul, as soon as posible i try with this schemas and code i will come back. The difference between total_price and price, was an error during the screenshots, currently is already fixed.
Reading your code, that might work, my problem is related to typescript type and Prisma but i will try.

**Reply by Marcos-Iorio** [reply]
So, I test the code that you send to me and the problem persists.
I made a postman call, and the orderInfo is arriving correctly but for a reason Prisma doesn't like the IProducts[] interface. Maybe you can investigate what the errors mean? i didn't found anything.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
